### DISCUSSION TITLE
return 403 from libresbc or user agent

### DISCUSSION THREAD

**Discussion by rahman77889** [hnimminh/libresbc#62] [Q&A] [answered]
I feel confusing, please help

I have configured all setup in libreSBC

call flow : 
invite
return 407 : proxy authentication required
invite
return 403 : Forbidden

this kind of answer forbidden is from SBC or from the user agent ?

if from SBC, which part of API to configure it ?

if from the user agent is that because of username/password gateway or no?

my SBC IP is 192.168.60.140
route IP to SBC is 192.168.60.129
IP user agent is  10.100.0.20 and 10.110.0.20

but if we see the pcap it not yet communicate to user agent 

this is screenshot pcap files
![image](https://user-images.githubusercontent.com/4687647/182764096-9d05a9d0-0da8-4823-9803-5364773d75ff.png)

Thank you. very much

**Comment by hnimminh** [comment]
I did not get your question, for configuration please follow https://github.com/hnimminh/libresbc/wiki

**Reply by bilbobg** [reply]
If I understood your problem right - there is no communication between the libresbc and the clients, please update the 'whiteset' firewall rules with your clients IP addresses.

https://your_libresbc_ip_addr:8443/apidocs#operation/update_fwset_libreapi_base_firewall__nameset__patch

**Reply by rahman77889** [reply]
> If I understood your problem right - there is no communication between the libresbc and the clients, please update the 'whiteset' firewall rules with your clients IP addresses.
> 
> https://your_libresbc_ip_addr:8443/apidocs#operation/update_fwset_libreapi_base_firewall__nameset__patch

all IP already set to whiteset on firewall and also whiteips on module service

**Reply by rahman77889** [reply]
> I did not get your question, for configuration please follow https://github.com/hnimminh/libresbc/wiki

this is call flow process

![image](https://user-images.githubusercontent.com/4687647/182903092-477df8c7-e00c-4c6f-90b3-941761e473aa.png)


I dont't know which side that return 403 is that from libresbc or from user agent

but until now, I still don't know if show sngrep call flow to user agent like below is that user agent already connect or no ? 

![image](https://user-images.githubusercontent.com/4687647/182903282-d145ee1b-b37a-4e37-8081-60352db281ba.png)


thank you for your response

**Reply by sakurfull** [reply]
@rahman77889 Can you sow me your example of re-routeip. I have same problem. Thanks

**Comment by rahman77889** [comment, marked as answer]
After re-routing IP segment then finnaly go answer from MSC / Core

Thank's for your response @hnimminh & @bilbobg

but there's some another question will create new discussion topics

**Comment by sakurfull** [comment]
@rahman77889 You can help me with an example of Trunk Sip configuration for communnicate with LibreSBC? Actually use This

type=peer
qualify=yes
port=5060
insecure=port,invite
host=
dtmfmode=rfc2833
disallow=all
context=from-internal
allow=alaw&ulaw
secret=libre@secret
